lib/facter/easyrsa.rb

```diff
@@ -0,0 +1,57 @@
+Facter.add(:easyrsa) do
+  confine :kernel => 'Linux'
+  setcode do
+    version = ''
+    binaryv2 = ''
+    binaryv3 = ''
+    operatingsystem = Facter.value(:operatingsystem)
+    operatingsystemrelease = Facter.value(:operatingsystemrelease)
+
+    case operatingsystem
+    when /RedHat|CentOS/
+      binaryv2 = '/usr/share/easy-rsa/2.0/pkitool'
+      binaryv3 = '/usr/share/easy-rsa/3/easyrsa'
+    when /Ubuntu|Debian/
+      case operatingsystemrelease
+      when /8|9|16.04|18.04/
+        binaryv2 = '/usr/share/easy-rsa/pkitool'
+        binaryv3 = '/usr/share/easy-rsa/easyrsa'
+      else
+        binaryv2 = '/usr/share/doc/openvpn/examples/easy-rsa/2.0/pkitool'
+        binaryv3 = '/usr/share/doc/openvpn/examples/easy-rsa/3.0/easyrsa'
+      end
+    when /Archlinux/
+      binaryv2 = '/usr/share/easy-rsa/2.0/pkitool'
+      binaryv3 = '/usr/share/easy-rsa/3/easyrsa'
+    when /Amazon/
+      binaryv2 = '/usr/share/easy-rsa/2.0/pkitool'
+      binaryv3 = '/usr/share/easy-rsa/3/easyrsa'
+    when /FreeBSD/
+      binaryv2 = '/usr/local/share/easy-rsa/pkitool'
+      binaryv3 = '/usr/local/share/easy-rsa/easyrsa'
+    end
+
+    if File.exist? binaryv2
+      Open3.popen2("#{binaryv2} --help") do |stdin, stdout, status_thread|
```

Comment:
Puppet has some tooling to check if a binary is within `$PATH` and can also execute it:
https://github.com/voxpupuli/puppet-rabbitmq/blob/d6a216f985e421e81ac3e66dcfcd59faaada9da1/lib/facter/erl_ssl_path.rb#L7-L8

Reply:
Nice. Will be used.